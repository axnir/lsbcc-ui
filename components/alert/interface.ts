export type Type = 'info' | 'positive' | 'negative' | 'warning';
export type TypeMap = Record<Type, string>;

export interface AlertProps {
  /**
   * Set this to change alert type
   * @default info
   */
  type?: 'info' | 'positive' | 'negative' | 'warning';
}