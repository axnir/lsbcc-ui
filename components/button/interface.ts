import { ButtonHTMLAttributes, MouseEventHandler } from 'react';

export type Size = 'large' | 'middle' | 'small';
export type Type = 'default' | 'primary';

export interface ButtonProps {
  size?: Size;
  type?: Type;
  disabled?: boolean;
  className?: string;
  onClick?: MouseEventHandler<HTMLElement>;
}
