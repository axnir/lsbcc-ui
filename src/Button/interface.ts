import type { MouseEventHandler } from 'react';

export interface ButtonProps {
  /**
   * @description  可选，按钮大小
   */
  size?: 'large' | 'middle' | 'small';
  /**
   * @description  可选，按钮的类型
   */
  type?: 'default' | 'primary';
  /**
   * @description  可选，按钮失效状态
   */
  disabled?: boolean;
  /**
   * @description  可选，组件类名
   */
  className?: string;
  /**
   * @description  可选，点击事件的回调
   */
  onClick?: MouseEventHandler<HTMLElement>;
}
