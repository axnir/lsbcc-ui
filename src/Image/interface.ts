import type { ImgHTMLAttributes, ReactNode } from 'react';

export interface ImageProps extends ImgHTMLAttributes<HTMLImageElement> {
  /**
   * @description      图片地址
   */
  src: string;
  /**
   * @description      可选，加载过程占位元素
   */
  fallback?: string;
  /**
   * @description      可选，加载失败的兜底图片
   */
  loader?: ReactNode;
}
