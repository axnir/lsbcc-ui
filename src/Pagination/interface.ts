import { MouseEventHandler } from 'react';

export interface PaginationProps {
  /**
   * @description      数据总数
   */
  pageSize: number;
  /**
   * @description      每页条数
   */
  total?: number;
  /**
   * @description      当前页数
   */
  current: number;
  /**
   * @description      页码改变的回调，参数是改变后的页码
   */
  onChange(page: number): void;
}

export type OmitPageProps = {
  onClick: MouseEventHandler<HTMLElement>;
};
