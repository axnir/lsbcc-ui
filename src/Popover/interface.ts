export interface PopoverProps {
  /**
   * @description      卡片标题
   */
  title: string;
  /**
   * @description      卡片内容
   */
  content: React.ReactNode;
  /**
   * @description      卡片位置
   */
  placement?: 'top' | 'right' | 'bottom' | 'left';
}
