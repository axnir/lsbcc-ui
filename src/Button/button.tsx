import React from 'react';
import classNames from 'classnames';
import type { FC } from 'react';
import type { ButtonProps } from './interface';

const prefixCls = 'lsbcc-button';

const Button: FC<ButtonProps> = ({
  type = 'default',
  size = 'middle',
  disabled,
  children,
  className,
  onClick,
  ...buttonProps
}) => {
  const classes = classNames(prefixCls, className, {
    primary: type === 'primary',
    large: size === 'large',
    small: size === 'small',
    disabled,
  });

  const handleClick = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
    if (disabled) {
      e.preventDefault();
      return;
    }
    onClick && (onClick as React.MouseEventHandler<HTMLButtonElement>)(e);
  };

  return (
    <button {...buttonProps} type="button" className={classes} onClick={handleClick}>
      <span className="btn-name">{children}</span>
    </button>
  );
};

export default Button;
