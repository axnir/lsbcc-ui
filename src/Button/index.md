---
nav:
  title: 组件
  path: /components
  order: 2
group:
  title: 组件
---

# Button 按钮

开始一个即时操作。

## 代码演示

### 基本用法

<code src="./demo/index.tsx"></code>

<API src="./button.tsx"></API>
