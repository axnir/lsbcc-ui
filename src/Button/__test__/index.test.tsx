import React from 'react';
import { render } from '@testing-library/react';
import Button from '../button';
import { ButtonProps } from '../interface';

describe('<Popover />', () => {
  test('should render default', () => {
    const { container } = render(<Button>button</Button>);
    expect(container).toMatchSnapshot();
  });

  test('should render button with type', () => {
    const types: ButtonProps['type'][] = ['default', 'primary'];

    const { getByText } = render(
      <>
        {types.map((t) => (
          <Button key={t} type={t}>
            {t}
          </Button>
        ))}
      </>,
    );

    types.forEach((t) => {
      t && expect(getByText(t)).toMatchSnapshot();
    });
  });

  test('should render button with size', () => {
    const types: ButtonProps['size'][] = ['large', 'middle', 'small'];

    const { getByText } = render(
      <>
        {types.map((t) => (
          <Button key={t} size={t}>
            {t}
          </Button>
        ))}
      </>,
    );

    types.forEach((t) => {
      t && expect(getByText(t)).toMatchSnapshot();
    });
  });

  test('should render button with disabled', () => {
    const { container } = render(
      <>
        <Button disabled>disabled</Button>
      </>,
    );

    expect(container).toMatchSnapshot();
  });
});
