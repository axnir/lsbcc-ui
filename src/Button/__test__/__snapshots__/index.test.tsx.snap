// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popover /> should render button with disabled 1`] = `
<div>
  <button
    class="lsbcc-button disabled"
    type="button"
  >
    <span
      class="btn-name"
    >
      disabled
    </span>
  </button>
</div>
`;

exports[`<Popover /> should render button with size 1`] = `
<span
  class="btn-name"
>
  large
</span>
`;

exports[`<Popover /> should render button with size 2`] = `
<span
  class="btn-name"
>
  middle
</span>
`;

exports[`<Popover /> should render button with size 3`] = `
<span
  class="btn-name"
>
  small
</span>
`;

exports[`<Popover /> should render button with type 1`] = `
<span
  class="btn-name"
>
  default
</span>
`;

exports[`<Popover /> should render button with type 2`] = `
<span
  class="btn-name"
>
  primary
</span>
`;

exports[`<Popover /> should render default 1`] = `
<div>
  <button
    class="lsbcc-button"
    type="button"
  >
    <span
      class="btn-name"
    >
      button
    </span>
  </button>
</div>
`;
